api/v1alpha1/storageconsumer_types.go

```diff
@@ -100,7 +100,9 @@ type StorageConsumerStatus struct {
 	// Information of storage client received from consumer
 	Client                       ClientStatus                `json:"client,omitempty"`
 	ResourceNameMappingConfigMap corev1.LocalObjectReference `json:"resourceNameMappingConfigMap,omitempty"`
-	OnboardingTicketSecret       corev1.LocalObjectReference `json:"onboardingTicketSecret,omitempty"`
+	// +optional
+	// +nullable
+	OnboardingTicketSecret *corev1.LocalObjectReference `json:"onboardingTicketSecret,omitempty"`
```

Comment:
How come this need to be a pointer while ClientStatus struct does not? What is the difference?

Reply:
this allows us to set `.status.onboardingTicketSecret = nil` after onboard which removes the field from getting serialized.